x-pack/legacy/plugins/apm/public/components/shared/MetadataTable/index.tsx

```diff
@@ -9,83 +9,113 @@ import {
   EuiFlexItem,
   EuiIcon,
   EuiSpacer,
-  EuiTitle
+  EuiTitle,
+  EuiFieldSearch
 } from '@elastic/eui';
-import React from 'react';
-import { get, pick, isEmpty } from 'lodash';
-import { EuiText } from '@elastic/eui';
+import React, { useState, useCallback } from 'react';
 import { i18n } from '@kbn/i18n';
-import { DottedKeyValueTable } from '../DottedKeyValueTable';
+import { isEmpty } from 'lodash';
+import { EuiText } from '@elastic/eui';
 import { ElasticDocsLink } from '../../shared/Links/ElasticDocsLink';
 import { Section as SectionType } from './sections';
 import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
 import { APMError } from '../../../../typings/es_schemas/ui/APMError';
 import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { HeightRetainer } from '../HeightRetainer';
+import { Section } from './Section';
+import { history } from '../../../utils/history';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+import { useLocation } from '../../../hooks/useLocation';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { filterSections } from './helper';
 
-type Item = Transaction | APMError | Span;
+export type Item = Transaction | APMError | Span;
 
 interface Props {
   item: Item;
   sections: SectionType[];
 }
 
-const filterSections = (sections: SectionType[], item: Item) =>
-  sections
-    .map(section => {
-      const data: Record<string, unknown> = get(item, section.key);
-      return {
-        ...section,
-        data: section.properties ? pick(data, section.properties) : data
-      };
-    })
-    .filter(({ required, data }) => required || !isEmpty(data));
-
 export function MetadataTable({ item, sections }: Props) {
-  const filteredSections = filterSections(sections, item);
+  const location = useLocation();
+  const { urlParams } = useUrlParams();
+  const { filter = '' } = urlParams;
+  const [filteredValue, setFilteredValue] = useState(filter || '');
+  const [filteredSections, setFilteredSections] = useState(
+    filterSections(sections, item, filter)
+  );
+
+  const onSearchChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const value = e.target.value.trim().toLowerCase();
+      setFilteredValue(value);
```

Comment:
In looks like this value is stored in multiple states: both in local react state as well as in the url. Better to just keep it in the url.